python-sdk/rebuff/detect_pi_vectorbase.py

```diff
@@ -0,0 +1,66 @@
+from typing import Dict, Union
+from langchain.vectorstores.pinecone import Pinecone
+from langchain.embeddings.openai import OpenAIEmbeddings
+import pinecone
+
+
+# https://api.python.langchain.com/en/latest/vectorstores/langchain.vectorstores.pinecone.Pinecone.html
+def detect_pi_using_vector_database(
+    input: str, similarity_threshold: float, vector_store: Pinecone
+) -> Union[Dict[str, int], str]:
+    try:
+        top_k = 20
+        results = vector_store.similarity_search_with_score(input, top_k)
+
+        top_score = 0
+        count_over_max_vector_score = 0
+
+        for _, score in results:
+            if score is None:
+                continue
+
+            if score > top_score:
+                top_score = score
+
+            if score >= similarity_threshold and score > top_score:
+                count_over_max_vector_score += 1
+
+        vector_score = {
+            "top_score": top_score,
+            "count_over_max_vector_score": count_over_max_vector_score,
+            "error": None,
+        }
+
+        return vector_score
+
+    except Exception as error:
+        vector_score = {
+            "top_score": None,
+            "count_over_max_vector_score": None,
+            "error": error,
```

Comment:
Since this function is not a web API, I'd recommend just raising an exception instead of including that as an field in the dictionary.

Reply:
Makes sense, thanks. Updated this and other instances (of adding error as a field) to raising exceptions.